src/types.rs

```diff
@@ -1515,6 +1582,10 @@ impl TryFrom<api::ReadSessionResponse> for ReadOutput {
     }
 }
 
+/// Name of a basin.
+///
+/// Must be between 8 and 48 characters in length. Must comprise lowercase
+/// letters, numbers, and hyphens. Cannot begin or end with a hyphen.
 #[derive(Debug, Clone)]
 pub struct BasinName(String);
 
```

Comment:
ah this duplicates with what sync_docs syncs on CreateBasinRequest

Reply:
Some duplication is inevitable I suppose. The `CreateBasinRequest` has:

```rust
    /// Basin name, which must be globally unique. It can be omitted to let the service assign a unique name.
```

which is irrelevant.